### PR TITLE
fix(notion): pages resource returns only meta info

### LINKED ISSUE #413: Add pages resource to notion
### Source name

Notion

### Describe the data you'd like to see

I want to see page contents. Multiple of our users mentioned that's what they need for ML cases.

See filter here, 
https://github.com/dlt-hub/verified-sources/blob/master/sources/notion/__init__.py#L33

Please group all pages under a single resource, and add the page(document id as a column, so we have a simple access point for the pages. Do not create a table per page as we do with databases (there it makes sense to have separation)

### Are you a dlt user?

Yes, I'm already a dlt user.

### Do you ready to contribute this extension?

No.

### dlt destination

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by IlyaFaer**
Closes #413

### CODE REVIEW

**Comment by IlyaFaer** [comment]
@adrianbr, the thing is - I don't have straight access to the Notion account, so I can't always say if it's content or just metadata - some fields look like what's needed to me.

With new changes, it returns blocks, let's see:
```
[
    {
        "object": "block",
        "id": "3d28f2cb-dd86-4207-b25c-f36af16e2f8d",
        "parent": {
            "type": "page_id",
            "page_id": "29298248-6067-4332-b6db-ca516d4d9af3",
        },
        "created_time": "2023-05-08T16:10:00.000Z",
        "last_edited_time": "2023-05-08T16:10:00.000Z",
        "created_by": {"object": "user", "id": "4625a1c7-b3f5-44d2-a0ed-c50f33882402"},
        "last_edited_by": {
            "object": "user",
            "id": "4625a1c7-b3f5-44d2-a0ed-c50f33882402",
        },
        "has_children": False,
        "archived": False,
        "in_trash": False,
        "type": "heading_1",
        "heading_1": {
            "rich_text": [
                {
                    "type": "text",
                    "text": {"content": "Background info", "link": None},
                    "annotations": {
                        "bold": False,
                        "italic": False,
                        "strikethrough": False,
                        "underline": False,
                        "code": False,
                        "color": "default",
                    },
                    "plain_text": "Background info",
                    "href": None,
                }
            ],
            "is_toggleable": False,
            "color": "default",
        },
    },
    {
        "object": "block",
        "id": "944d2062-ba72-43cc-96f3-577af7c559c8",
        "parent": {
            "type": "page_id",
            "page_id": "29298248-6067-4332-b6db-ca516d4d9af3",
        },
        "created_time": "2023-05-08T16:10:00.000Z",
        "last_edited_time": "2023-05-08T16:10:00.000Z",
        "created_by": {"object": "user", "id": "4625a1c7-b3f5-44d2-a0ed-c50f33882402"},
        "last_edited_by": {
            "object": "user",
            "id": "4625a1c7-b3f5-44d2-a0ed-c50f33882402",
        },
        "has_children": False,
        "archived": False,
        "in_trash": False,
        "type": "bulleted_list_item",
        "bulleted_list_item": {
            "rich_text": [
                {
                    "type": "text",
                    "text": {"content": "...", "link": None},
                    "annotations": {
                        "bold": False,
                        "italic": False,
                        "strikethrough": False,
                        "underline": False,
                        "code": False,
                        "color": "default",
                    },
                    "plain_text": "...",
                    "href": None,
                }
            ],
            "color": "default",
        },
    },
    {
        "object": "block",
        "id": "f7eb3202-9ba5-4946-8484-134f58d95f32",
        "parent": {
            "type": "page_id",
            "page_id": "29298248-6067-4332-b6db-ca516d4d9af3",
        },
        "created_time": "2023-05-08T16:10:00.000Z",
        "last_edited_time": "2023-05-08T16:10:00.000Z",
        "created_by": {"object": "user", "id": "4625a1c7-b3f5-44d2-a0ed-c50f33882402"},
        "last_edited_by": {
            "object": "user",
            "id": "4625a1c7-b3f5-44d2-a0ed-c50f33882402",
        },
        "has_children": False,
        "archived": False,
        "in_trash": False,
        "type": "bulleted_list_item",
        "bulleted_list_item": {
            "rich_text": [
                {
                    "type": "text",
                    "text": {"content": "...", "link": None},
                    "annotations": {
                        "bold": False,
                        "italic": False,
                        "strikethrough": False,
                        "underline": False,
                        "code": False,
                        "color": "default",
                    },
                    "plain_text": "...",
                    "href": None,
                }
            ],
            "color": "default",
        },
    },
    {
        "object": "block",
        "id": "5c2df167-fea8-463f-b880-05c9e322bbe5",
        "parent": {
            "type": "page_id",
            "page_id": "29298248-6067-4332-b6db-ca516d4d9af3",
        },
        "created_time": "2023-05-08T16:10:00.000Z",
        "last_edited_time": "2023-05-08T16:10:00.000Z",
        "created_by": {"object": "user", "id": "4625a1c7-b3f5-44d2-a0ed-c50f33882402"},
        "last_edited_by": {
            "object": "user",
            "id": "4625a1c7-b3f5-44d2-a0ed-c50f33882402",
        },
        "has_children": False,
        "archived": False,
        "in_trash": False,
        "type": "heading_1",
        "heading_1": {
            "rich_text": [
                {
                    "type": "text",
                    "text": {"content": "History of interactions", "link": None},
                    "annotations": {
                        "bold": False,
                        "italic": False,
                        "strikethrough": False,
                        "underline": False,
                        "code": False,
                        "color": "default",
                    },
                    "plain_text": "History of interactions",
                    "href": None,
                }
            ],
            "is_toggleable": False,
            "color": "default",
        },
    },
    {
        "object": "block",
        "id": "22afc43a-db33-4d01-9ada-9930d4f265cc",
        "parent": {
            "type": "page_id",
            "page_id": "29298248-6067-4332-b6db-ca516d4d9af3",
        },
        "created_time": "2023-05-08T16:10:00.000Z",
        "last_edited_time": "2023-05-08T16:10:00.000Z",
        "created_by": {"object": "user", "id": "4625a1c7-b3f5-44d2-a0ed-c50f33882402"},
        "last_edited_by": {
            "object": "user",
            "id": "4625a1c7-b3f5-44d2-a0ed-c50f33882402",
        },
        "has_children": False,
        "archived": False,
        "in_trash": False,
        "type": "bulleted_list_item",
        "bulleted_list_item": {
            "rich_text": [
                {
                    "type": "text",
                    "text": {"content": "...", "link": None},
                    "annotations": {
                        "bold": False,
                        "italic": False,
                        "strikethrough": False,
                        "underline": False,
                        "code": False,
                        "color": "default",
                    },
                    "plain_text": "...",
                    "href": None,
                }
            ],
            "color": "default",
        },
    },
    {
        "object": "block",
        "id": "4d76ff34-1649-437c-900f-32cc8e0d3cc4",
        "parent": {
            "type": "page_id",
            "page_id": "29298248-6067-4332-b6db-ca516d4d9af3",
        },
        "created_time": "2023-05-08T16:10:00.000Z",
        "last_edited_time": "2023-05-08T16:10:00.000Z",
        "created_by": {"object": "user", "id": "4625a1c7-b3f5-44d2-a0ed-c50f33882402"},
        "last_edited_by": {
            "object": "user",
            "id": "4625a1c7-b3f5-44d2-a0ed-c50f33882402",
        },
        "has_children": False,
        "archived": False,
        "in_trash": False,
        "type": "bulleted_list_item",
        "bulleted_list_item": {
            "rich_text": [
                {
                    "type": "text",
                    "text": {"content": "...", "link": None},
                    "annotations": {
                        "bold": False,
                        "italic": False,
                        "strikethrough": False,
                        "underline": False,
                        "code": False,
                        "color": "default",
                    },
                    "plain_text": "...",
                    "href": None,
                }
            ],
            "color": "default",
        },
    },
]
```

File: tests/notion/test_pages_resource.py [hunk 1]
```diff
@@ -36,20 +31,29 @@ def test_load_all_pages(destination_name: str):
...
+    }
+
+    assert loaded_tables == expected_tables
 
     with pipeline.sql_client() as client:
         with client.execute_query("SELECT * FROM notion_pages") as cur:
-            assert cur.fetchone()[1] in requested_pages
+            row = cur.fetchone()
+            assert row[0] == "block"
+            assert row[3] == page_id
```
Comment (IlyaFaer) [inline tests/notion/test_pages_resource.py:59]: Checking the resource type and id of the parent page

**Comment by rudolfix** [comment]
@adrianbr are you going to review it? if not please assign someone so we get exactly what you wanted :)

**Comment by IlyaFaer** [comment]
This PR is a bit stuck

**Comment by rudolfix** [comment]
@adrianbr are you able to verify that it does what you need? if not I'll approve it. because it LGTM!

**Review by rudolfix — APPROVED** [review]
LGTM! As far as I can understand it meets @adrianbr expectations

**Pull request merged**